app/commands/base_command.rb

```diff
@@ -16,6 +24,12 @@ def initialize(payload, event:, downstream: true)
       @downstream = downstream
     end
 
+  protected
+
+    def after_transaction_commit
+      self.class.callbacks = yield
```

Comment:
You could just capture the block here and call it later, right?  Like
```
def after_transaction_commit(&block)
  self.class.callbacks << block
end
```

and then iterate over the callbacks.